### DISCUSSION TITLE
Same transactions across multiple threads

### DISCUSSION THREAD

**Discussion by qizzzh** [googleapis/google-cloud-cpp#7522] [Q&A] [answered]
I'm using the C++ spanner client and make copies when necessary according to 

```
  * Instances of this class created via copy-construction or copy-assignment
  * share the underlying pool of connections. Access to these copies via multiple
  * threads is guaranteed to work. Two threads operating on the same instance of
  * this class is not guaranteed to work.
  * 
 ```

for thread safety. However I didn't find any documentation around whether it's safe to share the same transaction across multiple threads. Can someone confirm whether it's safe or not? Thank you!

**Comment by devbww** [comment, marked as answer]
Yes, a transaction can be used across multiple threads.  It seems there would need to be some synchronization to decide who commits and when, however.
